src/servers/src/http.rs

```diff
@@ -271,8 +272,9 @@ pub struct ApiState {
 }
 
 impl HttpServer {
-    pub fn new(sql_handler: SqlQueryHandlerRef) -> Self {
+    pub fn new(timeout: u64, sql_handler: SqlQueryHandlerRef) -> Self {
```

Comment:
Prefer to put the `timeout` arg as the second argument.

Reply:
As we might add more options in the future, what about passing the `HttpOptions` struct instead of just the timeout? Moving the `HttpOptions` to the `servers` crate would be better.

Reply:
> As we might add more options in the future, what about passing the `HttpOptions` struct instead of just the timeout? Moving the `HttpOptions` to the `servers` crate would be better.

Sounds reasonable 👍

Reply:
> As we might add more options in the future, what about passing the `HttpOptions` struct instead of just the timeout? Moving the `HttpOptions` to the `servers` crate would be better.

Agree. The only concern here is we have to expose HttpOptions to servers firstly.

Reply:
Currently `frontend::server::Services::start` initializes all servers and modifies their options via mutable setters (e.g.: `http_server.set_script_handler`), it already receives `FrontendOptions` as argument.

It feels natural to add more option-overriding logic here. For example (from my duplicate PR 🤣):

https://github.com/GreptimeTeam/greptimedb/blob/26d5bf2be9c3dac50cf93a5421e037dd9ae68f09/src/frontend/src/server.rs#L140-L144

As the number of setters grows, we can even refactor it into something like `HttpServerBuilder` 🤔.

Reply:
Thanks for your approach, whereas I did it by new() injecting